src/main/scala/cognite/spark/v1/SyncCursorCallback.scala

```diff
@@ -0,0 +1,88 @@
+package cognite.spark.v1
+
+import cats.effect.IO
+import cats.effect.unsafe.{IORuntime, IORuntimeConfig}
+import com.cognite.sdk.scala.sttp.GzipBackend
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import org.log4s.getLogger
+import sttp.client3.{SttpBackend, basicRequest}
+import sttp.client3.asynchttpclient.SttpClientBackendFactory
+import sttp.client3.asynchttpclient.cats.AsyncHttpClientCatsBackend
+import sttp.client3.circe.{asJson, circeBodySerializer}
+import sttp.model.{MediaType, Uri}
+
+import java.lang.Thread.UncaughtExceptionHandler
+import java.util.concurrent.Executors
+import scala.concurrent.ExecutionContext
+
+case class IncrementalCursor(name: String, value: String)
+case class IncrementalCursorResponse(name: String, value: String, updated: Boolean)
+
+object SyncCursorCallback {
+  @transient private val logger = getLogger
+  @transient private lazy val sttpBackend: SttpBackend[IO, Any] =
+    new GzipBackend[IO, Any](AsyncHttpClientCatsBackend.usingClient(SttpClientBackendFactory.create()))
```

Comment:
will need to check, but maybe we could reuse a common one or move this one to the others (in CdpConnector?)
additionally this default one doesn't have retries iirc

Reply:
On purpose not to have retries actually. Keep it as simple as it can be, as the downside of not committing the cursor is limited.

Reply:
any client should have at least some retries, there's no benefit in losing the call when there's a cheap chance to not lose it (and the call should be idempotent)

Reply:
I've now switched to using the retrying backend to have retries.